DB-5455: Advanced Custom Fields Add-on Docs (#545)
pantheon-systems/decoupled-kit-js: web/docs/Frontend Starters/Gatsby/Gatsby + WordPress/add-ons.md

---
id: 'gatsby-wordpress-add-ons'
title: 'Add-ons'
slug: '/frontend-starters/nextjs/gatsby/gatsby-wordpress/add-ons'
sidebar_position: 5
---

## Before You Begin

- Install and activate the
  [WP Advanced Custom Fields Plugin](https://wordpress.org/plugins/advanced-custom-fields/).

- Familiarize yourself with the
  [Create Pantheon Decoupled Kit CLI](https://www.npmjs.com/package/create-pantheon-decoupled-kit/).
  Any package manager can be used with the CLI **[npm, pnpm, yarn]**.

## Advanced Custom Fields WordPress Plugin

The Advanced Custom Fields WordPress plugin, or ACF, turns WordPress sites into
a fully-fledged content management system by giving you all the tools to do more

> with your data.

Reviewer: Two things:
1. If you take text directly from a source on the web, add an attribution to that source.
2. I want to avoid explaining what ACF is in our docs. I would expect someone reading this to already be familiar with ACF, and if not we can link to their docs so the user can learn about it before coming back here and integrating ACF with our starter via the add-on. So in that case, I would prefer a note in "Before You Begin" that explains "You should be familiar with WordPress Advanced Custom Fields" with a link to the docs. I don't think we should explain what it is here.

This section should explain what our add-on does, but we should avoid explaining ACF.

ACF lets you add, store, and display additional information about a piece of
content in WordPress. This information can easily be assessed through GraphQL
with simple adjustments to existing queries.

This add-on uses GraphQL to bring in a custom related content data type that was
created with ACF. After adding the `gatsby-wp-acf-addon`, our starter kit will
have a related content section that displays at the bottom of post detail pages,
if related posts are specified on the related WordPress post.

## Usage

### Adding the `gatsby-wp-acf-addon` to an Existing Project

1. Use the create command to initiate the cli with the `gatsby-wp-acf-addon`
   generator
   ```bash
   npm create pantheon-decoupled-kit gatsby-wp-acf-addon
   ```
1. When prompted for an output directory, set the output to the root directory

>    of your existing Gatsby Wordpress Starter

Reviewer: ```suggestion
   of your existing Gatsby WordPress Starter
```

---
id: 'gatsby-wordpress-add-ons'

> title: 'Add Ons'

Reviewer: ```suggestion
title: 'Add-ons'
```
'Add-ons' should be hyphenated in this case.

slug: '/frontend-starters/nextjs/gatsby/gatsby-wordpress/add-ons'
sidebar_position: 5
---

> ### Advanced Custom Fields

Reviewer: ```suggestion
## Advanced Custom Fields WordPress Plugin
```
use an h2 here, also probably doesn't hurt to use the full name

> Advanced Custom Fields turns WordPress sites into a fully-fledged content

Reviewer: ```suggestion
The Advanced Custom Fields WordPress plugin, or ACF, turns WordPress sites into a fully-fledged content
```
introduce the full name the first time with the abbreviation, so from here on you can just use 'ACF' instead of spelling out Advanced Custom Fields each time.

management system by giving you all the tools to do more with your data.

Advanced custom fields let you add, store, and display additional information
about a piece of content in WordPress. This information can easily be assessed
through GraphQL with simple adjustments to existing queries.

This add-on uses GraphQL to bring in a custom `related content` data type that

> was created with Advanced Custom Fields. Your starter kit will have a related

Reviewer: ```suggestion
was created with Advanced Custom Fields. After adding the `gatsby-wp-acf-addon`, our starter kit will have a related
```
More specificity here - otherwise it reads as though the starter kit has this already.

content section that displays at the bottom of post detail pages, if related
posts are specified on the related WordPress post.

#### Before You Begin

- Install and activate the
  [WP Advanced Custom Fields Plugin](https://wordpress.org/plugins/advanced-custom-fields/).

- Install the
  [Create Pantheon Decoupled Kit CLI](https://www.npmjs.com/package/create-pantheon-decoupled-kit/).

> 

Reviewer: "Before You Begin" should go at the top, also as an h2.

there should be no need to actually install the decoupled-kit cli. `npm create pantheon-decoupled-kit` (or pnpm create) pull the latest binary from the registry and execute it. We should probably stick with `npm create` for these docs, or mention that it will work with 'your preferred package manager (npm, pnpm, yarn)'

Reviewer: @mitchellmarkoff Going back to this for a second – I am realizing this doc needs to be structured slightly differently because it will relate to multiple add-ons. I'm not sure this will be the structure we use going forward but for now I guess it makes sense.
Because of that, the `h2`s here should be the addon name and under it will be the description. Then `h3`s under that should start each section like Before You Begin, Usage, and so on. I've pushed up this change, let me know if it makes sense to you.

Author: Makes total sense. I was actually wondering how formatting would work when the time would come to add to this doc, I should have started that conversation earlier in the process. LGTM!

> #### Instillation Through CLI

Reviewer: ```suggestion
## Adding the `gatsby-wp-acf-addon`  to an Existing Project
```
'Installing' doesn't feel right to me, open to other options here. Also, this should either be a new section (h2), or be nested under a more generic heading like "Setup" or move "Usage" up to here as the generic section

1. Use the create command to initiate the cli

>    `pnpm create pantheon-decoupled-kit`

Reviewer: As mentioned, use `npm` here. Maybe in Before You Begin, mention that any (npm, pnpm, yarn) package manager will work.
You could include the add-on as an arg to skip step 2
`npm create pantheon-decoupled-kit gatsby-wp-acf-addon`

1. Select the `gatsby-wp-acf-addon` and go through the prompts. Set the output
   to the directory that your Gatsby Wordpress Starter is located

> 1. Once the app is generated, add

Reviewer: This line reads as though we are generating an app from 0, not adding an add-on. I think it's fine to describe both use cases, but it should be clear what the starting point is.

>    `https://acf-decoupled-wordpress-qa.pantheonsite.io/wp/graphql` as the

Reviewer: Not sure we want to publicize this URL 😬

   `WPGRAPHQL_URL` in `.env.development.local`
1. Start your project locally and observe the new related content section that
   displays at the bottom of post detail pages

#### Usage

##### Creating a Custom Field

In the Advanced Custom Fields plugin dashboard:

1. Select **Add New** next to field groups
1. Name your field group
1. Select a field type, field name, and field label for your custom field
1. Ensure **Show in GraphQL** is selected for your field
1. At the bottom of the form make sure **Show in GraphQL** is selected for your
   field group and that its generated field name is correct
1. Save your changes

##### Adding Custom Field Data

Once you have created a field group, values can be assigned to your new custom
fields via the Guttenberg editor.

1. Select and existing post or page, or create a new one.
1. At the bottom of your editor, you will see the new custom fields that have
   been created, fill them in with data of your choosing

> 1. Save and publish your content updates

Reviewer: Not sure this makes sense in this particular doc. I would expect that a user has some understanding of ACF from the WP side if they're using this add-on. This doc should concern itself more with how to fetch the data, not how to create it.
